sdk/iot/pnp/inc/az_iot_pnp_client.h

```diff
@@ -0,0 +1,332 @@
+ * @return #az_result
+ */
+AZ_NODISCARD az_result az_iot_pnp_client_telemetry_publish_topic_get(
+    az_iot_pnp_client const* client,
+    az_span component_name,
+    az_span content_type,
+    az_span encoding_type,
+    az_iot_pnp_client_telemetry_options const* options,
+    az_span mqtt_topic,
+    az_span* out_mqtt_topic);
+
+/*
+ *
+ * PnP Command APIs
+ *
+ */
+
+/**
+ * @brief Azure IoT PnP command subscription options.
+ *
+ */
+typedef struct az_iot_pnp_client_command_subscription_options
+{
+  struct
+  {
+    int _not_used;
+  } _internal;
+} az_iot_pnp_client_command_subscription_options;
+
+/**
+ * @brief Gets the default Azure IoT PnP command subscription options.
+ * @details Call this to obtain an initialized #az_iot_pnp_client_command_subscription_options
+ * structure that can be afterwards modified and passed to
+ * #az_iot_pnp_client_commands_subscribe_topic_filter_get.
+ *
+ * @return #az_iot_pnp_client_command_subscription_options.
+ */
+AZ_NODISCARD az_iot_pnp_client_command_subscription_options az_iot_pnp_client_command_subscription_options_default();
+
+/**
+ * @brief Gets the MQTT topic filter to subscribe to PnP commands.
+ *
+ * @param[in] client The #az_iot_pnp_client to use for this call.
+ * @param[in] options A reference to an #az_iot_pnp_client_command_subscription_options structure.
+ * Can be NULL.
+ * @param[in] mqtt_topic_filter An empty #az_span with sufficient capacity to hold the MQTT topic
+ *                              filter.
+ * @param[out] out_mqtt_topic_filter The output #az_span containing the MQTT topic filter.
+ * @return #az_result
+ */
+AZ_NODISCARD az_result az_iot_pnp_client_commands_subscribe_topic_filter_get(
+    az_iot_pnp_client const* client,
+    az_iot_pnp_client_command_subscription_options const* options,
+    az_span mqtt_topic_filter,
+    az_span* out_mqtt_topic_filter);
+
+/**
+ * @brief A command request received from the PnP service.
+ *
+ */
+typedef struct az_iot_pnp_client_command_request
+{
+  az_span request_id; /**< The request id.
+                       * @note The application must match the method request and method response. */
+  az_span component; /**< The component the command is targeted towards. */
+  az_span command_name; /**< The name of command to invoke */
+  az_span payload; /**< The body of the request for the command, stripped of any envelope data from
+                        MQTT payload */
+} az_iot_pnp_client_command_request;
+
+/**
+ * @brief Attempts to parse a received message's topic.
+ *
+ * @param[in] client The #az_iot_pnp_client to use for this call.
+ * @param[in] received_topic An #az_span containing the received topic.
+ * @param[in] received_payload An #az_span containing the received payload.
+ * @param[out] out_request If the message is a command request, this will contain the
+ *                         #az_iot_pnp_client_command_request.
+ * @return #az_result
```

Comment:
I think this probably goes for the hub docs as well but there should be documentation here about what to expect in the case that it IS a command request and in the case it IS NOT a command request. Is there a particular `az_result` that is returned when it is or is not? Is there something about the makeup of the returned `az_iot_pnp_client_command_request`?

Because if I'm a dev here, I could expect that `AZ_OK` is returned if it is a command OR if it was just successfully parsed and the notification that it is a command comes through the `out_request`. If that makes sense.

Reply:
Agreed, that merits special consideration in docs.  @CIPop  - I poked around IoT headers and I couldn't find this error code.  Is it declared yet?

Reply:
I think KEY az_result values should be specified here. See https://github.com/Azure/azure-sdk-for-c/blob/master/sdk/core/core/inc/az_json.h and search for az_json_token_get_boolean to see an example.